packages/components/Menu/src/Menu/useMenu.ts

```diff
@@ -61,6 +54,13 @@ const useMenuOpenState = (
 
   const state = isControlled ? open : openInternal;
 
+  // Due to how events get fired we get double notifications
+  // for the same event causing us to immediately reopen
+  // a menu when we close it. Adding in a delay to prevent
+  // this behavior.
```

Comment:
It looks like it was added [here](https://github.com/microsoft/fluentui-react-native/pull/1733) to avoid flickering behavior for the keyboard.

Reply:
Is there a way we can read `event.target` , and perhaps ignore the extra events that are coming from a different react view? Reason being I think React doesn't like timestamps with renders much, and also because a timestamp feels like a workaround for a workaround of an issue.